test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs

```diff
@@ -216,15 +232,68 @@ private Stream NormalizeNewLines(Stream inputStream)
 
                 if (expectedDesignTimePragmas != null)
                 {
-                    Assert.True(results.DesignTimeLineMappings != null); // Guard
+                    Assert.NotNull(results.DesignTimeLineMappings); // Guard
+#if GENERATE_BASELINES
+                    if (expectedDesignTimePragmas == null ||
+                        !Enumerable.SequenceEqual(expectedDesignTimePragmas, results.DesignTimeLineMappings))
+                    {
+                        var lineMappingFile = Path.ChangeExtension(outputFile, "lineMappings.cs");
+                        var lineMappingCode = GetDesignTimeLineMappingsCode(results.DesignTimeLineMappings);
+                        BaselineWriter.WriteBaseline(lineMappingFile, lineMappingCode);
+                    }
+#else
                     for (var i = 0; i < expectedDesignTimePragmas.Count && i < results.DesignTimeLineMappings.Count; i++)
                     {
                         Assert.Equal(expectedDesignTimePragmas[i], results.DesignTimeLineMappings[i]);
                     }
 
                     Assert.Equal(expectedDesignTimePragmas.Count, results.DesignTimeLineMappings.Count);
+#endif
+                }
+            }
+        }
+
+        private static string GetDesignTimeLineMappingsCode(IList<LineMapping> designTimeLineMappings)
+        {
+            var lineMappings = new StringBuilder();
+            lineMappings.AppendLine($"// !!! Do not check in. Instead paste content into test method. !!!");
+            lineMappings.AppendLine();
+
+            var indent = "            ";
+            lineMappings.AppendLine($"{ indent }var expectedLineMappings = new[]");
+            lineMappings.AppendLine($"{ indent }{{");
+            foreach (var lineMapping in designTimeLineMappings)
+            {
+                var innerIndent = indent + "    ";
+                var documentLocation = lineMapping.DocumentLocation;
+                var generatedLocation = lineMapping.GeneratedLocation;
+                lineMappings.AppendLine($"{ innerIndent }BuildLineMapping(");
```

Comment:
Not possible since `BuildLineMapping` is defined in a subclass.

Thanks for helping me see how badly out-of-sync the `RazorCodeGeneratorTest<TLanguage>` and `RazorCodeGeneratorTest` classes are with their containing files (`RazorChunkGeneratorTest.cs` and `CSharpRazorChunkGeneratorTest.cs`, respectively). `RazorCodeGeneratorTest` is also a fairly poor name for the subclass.